src/lando/main/models.py

```diff
@@ -1 +1,181 @@
+
+    status = models.CharField(
+        max_length=12,
+        choices=Status,
+        default=None,
+        null=True,  # TODO: should change this to not-nullable
+        blank=True,
+    )
+
+    # Text describing errors when status != LANDED.
+    error = models.TextField(default="", blank=True)
+
+    # Error details in a dictionary format, listing failed merges, etc...
+    # E.g. {
+    #    "failed_paths": [{"path": "...", "url": "..."}],
+    #     "reject_paths": [{"path": "...", "content": "..."}]
+    # }
+    error_breakdown = models.JSONField(null=True, blank=True, default=dict)
+
+    # LDAP email of the user who requested transplant.
+    requester_email = models.CharField(max_length=255)
+
+    # Lando's name for the repository.
+    repository_name = models.CharField(max_length=255)
+
+    # URL of the repository revisions are to land to.
+    repository_url = models.TextField(default="")
+
+    # Identifier for the most descendent commit created by this landing.
+    landed_commit_id = models.TextField(default="")
+
+    # Number of attempts made to complete the job.
+    attempts = models.IntegerField(default=0)
+
+    # Priority of the job. Higher values are processed first.
+    priority = models.IntegerField(default=0)
+
+    # Duration of job from start to finish
+    duration_seconds = models.IntegerField(default=0)
+
+    # Identifier of the published commit which this job should land on top of.
+    target_commit_hash = models.TextField(default="")
+
+    revisions = models.ManyToManyField(Revision)  # TODO: order by index
+
+    @classmethod
+    def job_queue_query(
+        cls,
+        repositories: Optional[Iterable[str]] = None,
+        grace_seconds: int = DEFAULT_GRACE_SECONDS,
+    ) -> QuerySet:
+        """Return a query which selects the queued jobs.
+
+        Args:
+            repositories (iterable): A list of repository names to use when filtering
+                the landing job search query.
+            grace_seconds (int): Ignore landing jobs that were submitted after this
+                many seconds ago.
+        """
+        applicable_statuses = (
+            cls.Status.SUBMITTED,
+            cls.Status.IN_PROGRESS,
+            cls.Status.DEFERRED,
+        )
+        q = cls.objects.filter(status__in=applicable_statuses)
+
+        if repositories:
+            q = q.filter(repository_name__in=(repositories))
+
+        if grace_seconds:
+            now = datetime.datetime.now(datetime.timezone.utc)
+            grace_cutoff = now - datetime.timedelta(seconds=grace_seconds)
+            q = q.filter(created_at__lt=grace_cutoff)
+
+        # Any `Status.IN_PROGRESS` job is first and there should
+        # be a maximum of one (per repository). For
+        # `Status.SUBMITTED` jobs, higher priority items come first
+        # and then we order by creation time (older first).
+        q = q.order_by("-status").order_by("-priority").order_by("created_at")
```

Comment:
Is there a way to do `order_by` without using strings? Referencing the field values somehow?

Reply:
This is the standard way of ordering queries in Django (though I just realized I meant to but forgot to combine this into one statement instead of three -- will make this change.